test/v200/utils/api/test_utils.go

```diff
@@ -0,0 +1,173 @@
+
+	// Read the created yaml file, ready for converison to json
+	devfileData, err := ioutil.ReadFile(devfile)
+	if err != nil {
+		commonUtils.LogErrorMessage(fmt.Sprintf("  FAIL: schema : unable to read %s: %v", devfile, err))
+		return err
+	}
+
+	// Convert the yaml file to json
+	devfileDataAsJSON, err := yaml.YAMLToJSON(devfileData)
+	if err != nil {
+		commonUtils.LogErrorMessage(fmt.Sprintf("  FAIL : %s : schema : failed to convert to json : %v", devfile, err))
+		return err
+	}
+
+	validationErr := schemaFile.Schema.Validate(bytes.NewReader(devfileDataAsJSON))
+	if validationErr != nil {
+		if len(expectedMessage) > 0 {
+			if !strings.Contains(validationErr.Error(), expectedMessage) {
+				err = errors.New(commonUtils.LogErrorMessage(fmt.Sprintf("  FAIL : schema : %s : Did not fail as expected : %s  got : %v", devfile, expectedMessage, validationErr)))
+			} else {
+				commonUtils.LogInfoMessage(fmt.Sprintf("PASS: schema :  Expected Error received : %s", expectedMessage))
+			}
+		} else {
+			err = errors.New(commonUtils.LogErrorMessage(fmt.Sprintf("  FAIL : schema : %s : Did not pass as expected, got : %v", devfile, validationErr)))
+		}
+	} else {
+		if len(expectedMessage) > 0 {
+			err = errors.New(commonUtils.LogErrorMessage(fmt.Sprintf("  FAIL : schema : %s :  was valid - Expected Error not found : %v", devfile, validationErr)))
+		} else {
+			commonUtils.LogInfoMessage(fmt.Sprintf("  PASS : schema : %s : devfile was valid.", devfile))
+		}
+	}
+	return err
+}
+
+// GetSchema downloads and saves a schema from the provided url
+func GetSchema(schemafile string) (SchemaFile, error) {
+
+	var err error
+	schemaFile, found := schemas[schemafile]
+	if !found {
+
+		schemaFile = SchemaFile{}
+
+		// Prepare the schema file
+		compiler := jsonschema.NewCompiler()
+		compiler.Draft = jsonschema.Draft7
+		schemaFile.Schema, err = compiler.Compile(schemafile)
+		if err != nil {
+			commonUtils.LogErrorMessage(fmt.Sprintf("FAIL : Failed to compile schema  %v", err))
+		} else {
+			commonUtils.LogInfoMessage(fmt.Sprintf("Schema compiled from file: %s)", schemafile))
+			schemas[schemafile] = schemaFile
+		}
+	}
+	return schemaFile, err
+}
+
+// WriteDevfile creates a devfile on disk for use in a test.
+func writeDevfile(devfile *commonUtils.TestDevfile) error {
+	var err error
+
+	fileName := devfile.FileName
+	if !strings.HasSuffix(fileName, ".yaml") {
+		fileName += ".yaml"
+	}
+
+	commonUtils.LogInfoMessage(fmt.Sprintf("Marshall and write devfile %s", devfile.FileName))
+
+	c, marshallErr := yaml.Marshal(&(devfile.SchemaDevFile))
+
+	if marshallErr != nil {
+		err = errors.New(commonUtils.LogErrorMessage(fmt.Sprintf("Marshall devfile %s : %v", devfile.FileName, marshallErr)))
+	} else {
+		err = ioutil.WriteFile(fileName, c, 0644)
+		if err != nil {
+			commonUtils.LogErrorMessage(fmt.Sprintf("Write devfile %s : %v", devfile.FileName, err))
+		}
+	}
+	return err
+}
+
+// validateDevfile check the provided defile against the schema
```

Comment:
a minor typo need to be fixed
```suggestion
// validateDevfile check the provided devfile against the schema
```